src/main/java/org/worshipsongs/importer/TopicDao.java

```diff
@@ -8,55 +8,60 @@
 /**
  * Created by Pitchu on 10/18/2015.
  */
-public class TopicDao {
-    public int getTopicId(Connection connection, String topic)
+public class TopicDao implements ITopicDao
+{
+    private Connection connection;
+    public TopicDao(Connection connection)
+    {
+        this.connection = connection;
+    }
+
+    public Topic getTopic(Topic topic)
     {
         int id = 0;
         try {
             Statement statement = null;
             statement = connection.createStatement();
-            ResultSet resultSet = statement.executeQuery( "SELECT * FROM TOPICS where name = '" + topic + "';" );
-            if(resultSet.next()) {
+            ResultSet resultSet = statement.executeQuery("SELECT * FROM TOPICS where name = '" + topic.getTopic() + "';");
+            if (resultSet.next()) {
                 id = resultSet.getInt("id");
             }
             resultSet.close();
             statement.close();
+        } catch (Exception e) {
+            System.out.println("Exception:" + e);
         }
-        catch (Exception e) {
-            System.out.println("Exception:"+e);
-        }
-        return id;
+        topic.setId(id);
+        return topic;
     }
 
-    public boolean insertTopicSongs(Connection connection, Topic topic, int songId)
+    public boolean insertTopicSongs(Song song)
     {
         try {
             String query = "insert into songs_topics (song_id, topic_id) values (?, ?)";
             PreparedStatement preparedStatement = connection.prepareStatement(query);
-            preparedStatement.setInt(1, songId);
-            preparedStatement.setInt(2, topic.getId());
+            preparedStatement.setInt(1, song.getId());
+            preparedStatement.setInt(2, song.getTopic().getId());
             preparedStatement.executeUpdate();
             preparedStatement.close();
             return true;
-        }
-        catch (Exception e) {
-            e.printStackTrace();
+        } catch (Exception e) {
+            System.out.println("Exception:" + e);
             return false;
         }
     }
 
-    public int insertTopic(Connection connection, String topic)
+    public Topic insertTopic(Topic topic)
```

Comment:
Rename the method as "create". "create" methods normally do not return anything.